node/root_peers.js

```diff
@@ -38,6 +39,10 @@ function TChannelRootPeers(channel, options) {
     self.allocPeerEvent = self.defineEvent('allocPeer');
     self.peerOptions = self.options.peerOptions || {};
     self.preferOutgoing = self.options.preferOutgoing;
+    self.preferIncoming = self.options.preferIncoming;
+    assert(!self.preferOutgoing || !self.preferIncoming,
+        'preferOutgoing and preferIncoming cannot be both set');
+
```

Comment:
In earnest, `options.preferConnectionDirection: "incoming" / "outgoing" / null` will DRY up much more easily, regardless of whether we eventually DRY up the three score strategies into one.